lib/internal/test_runner/runner.js

```diff
@@ -614,6 +625,65 @@ function run(options = kEmptyObject) {
     });
   }
   validateOneOf(isolation, 'options.isolation', ['process', 'none']);
+  if (coverage != null) {
+    validateBoolean(coverage, 'options.coverage');
+  }
+  if (coverageExcludeGlobs != null) {
+    if (!coverage) {
+      throw new ERR_INVALID_ARG_VALUE(
+        'options.coverageExcludeGlobs',
+        coverageExcludeGlobs,
+        'is only supported when coverage is enabled',
+      );
+    }
+    if (!ArrayIsArray(coverageExcludeGlobs)) {
+      coverageExcludeGlobs = [coverageExcludeGlobs];
+    }
+    validateStringArray(coverageExcludeGlobs, 'options.coverageExcludeGlobs');
+  }
+  if (coverageIncludeGlobs != null) {
+    if (!coverage) {
+      throw new ERR_INVALID_ARG_VALUE(
+        'options.coverageIncludeGlobs',
+        coverageIncludeGlobs,
+        'is only supported when coverage is enabled',
+      );
+    }
+    if (!ArrayIsArray(coverageIncludeGlobs)) {
+      coverageIncludeGlobs = [coverageIncludeGlobs];
+    }
+    validateStringArray(coverageIncludeGlobs, 'options.coverageIncludeGlobs');
+  }
+  if (lineCoverage != null) {
+    if (!coverage) {
+      throw new ERR_INVALID_ARG_VALUE(
+        'options.lineCoverage',
+        lineCoverage,
+        'is only supported when coverage is enabled',
+      );
+    }
+    validateInteger(lineCoverage, 'options.lineCoverage', 0, 100);
+  }
+  if (branchCoverage != null) {
+    if (!coverage) {
+      throw new ERR_INVALID_ARG_VALUE(
+        'options.branchCoverage',
+        branchCoverage,
+        'is only supported when coverage is enabled',
+      );
+    }
+    validateInteger(branchCoverage, 'options.branchCoverage', 0, 100);
+  }
+  if (functionCoverage != null) {
+    if (!coverage) {
+      throw new ERR_INVALID_ARG_VALUE(
+        'options.functionCoverage',
+        functionCoverage,
+        'is only supported when coverage is enabled',
+      );
+    }
+    validateInteger(functionCoverage, 'options.functionCoverage', 0, 100);
+  }
```

Comment:
non-blocking: Do we really need to throw in these cases? Can the options just be safely silently ignored if they aren't relevant? Or maybe just emit a warning?

It would be nice if we could reduce the duplication here. Can the check be moved into a separate utility function?

Reply:
If they must exist they could probably be simplified:
```suggestion
const coverageOptions = [
  { name: 'coverageExcludeGlobs', value: coverageExcludeGlobs, validate: validateStringArray, isArray: true },
  { name: 'coverageIncludeGlobs', value: coverageIncludeGlobs, validate: validateStringArray, isArray: true },
  { name: 'lineCoverage', value: lineCoverage, validate: validateInteger, range: [0, 100] },
  { name: 'branchCoverage', value: branchCoverage, validate: validateInteger, range: [0, 100] },
  { name: 'functionCoverage', value: functionCoverage, validate: validateInteger, range: [0, 100] }
];

for (const { name, value, validate, isArray, range } of coverageOptions) {
  if (value != null) {
    if (!coverage) {
      throw new ERR_INVALID_ARG_VALUE(
        `options.${name}`,
        value,
        `is only supported when coverage is enabled`
      );
    }
    let validatedValue = value;
    if (isArray && !ArrayIsArray(validatedValue)) {
      validatedValue = [validatedValue];
    }
    if (range) {
      validate(validatedValue, `options.${name}`, ...range);
    } else {
      validate(validatedValue, `options.${name}`);
    }
  }
}

```

Reply:
Heh, yeah, got distracted in the middle of reviewing and realized I had duplicated the comment.

Reply:
> Do we really need to throw in these cases?

I think we can simply ignore this, as that is the behavior for the CLI flag.
@cjihrig @MoLow  WDYT?

Reply:
I'm fine with ignoring them if they aren't relevant (we should probably document as such). We should keep the validation though.